salesagility/SuiteCRM
# Fix #5076 Smtp full log

## Description
Gets the correct, more informative, exception text from PHPMailer when sending email gives an error.
Let's the admin user see the full SMTP log in case of error.

## Motivation and Context
- SMTP is getting trickier with time, people are facing more restrictions from ISP's, like required certificates, or "From" address checking (must match account).
- our Email problems are compounded when it is not clear whose fault the error is: the SMTP server, the configuration, or some UI or database problem in SuiteCRM. This makes things much more transparent.

## How To Test This

Note: to send unsuccessful emails you can configure a wrong host name, or a wrong password.

1. Send **unsuccessful** emails from all these places, and expect a new message box with detailed error:

    a. Profile->down to the bottom->Settings->Mail Accounts->Mail Accounts (the top half of the screen)->ADD(or EDIT)

    b. Profile->down to the bottom->Settings->Mail Accounts->Outgoing SMTP (lower half, list box)->ADD (or in the list view, EDIT icon). Watch out, this part of SuiteCRM seems to be buggy, you might have trouble sending.

    c. Admin / Outbound email accounts, add

2. Send **successful** emails from all the above, the usual simple old message should appear.

3. Send **unsuccessful** emails from all other places, where my code doesn't do anything, so the expected result is that it doesn't break anything (i.e. just check there are no regressions):

    c. Sending from Compose window.

    d. Sending from Campaigns.

## Authentication data obfuscation:
When a full log is displayed, it should not contain the base64-encoded usernames and passwords of email accounts.

This is what the logs will look like:

    CLIENT -> SERVER: AUTH LOGIN
    SMTP -> get_lines(): $data is ""
    SMTP -> get_lines(): $str is "334 VXNlcm5hbWU6"
    SERVER -> CLIENT: 334 VXNlcm5hbWU6
    CLIENT -> SERVER: bWVAc2l0ZS5jb20=  <---------------
    SMTP -> get_lines(): $data is ""
    SMTP -> get_lines(): $str is "334 UGFzc3dvcmQ6"
    SERVER -> CLIENT: 334 UGFzc3dvcmQ6
    CLIENT -> SERVER: MTIzNDU2     <-----------------------
    SMTP -> get_lines(): $data is ""
    SMTP -> get_lines(): $str is "235 2.7.0 Accepted"

So, seeing `VXNlcm5hbWU6"` and `UGFzc3dvcmQ6`, it just means `User:` and `Password:`, but seeing other strings in place of those marked here with arrows should not happen, they should show as **---obfuscated---**.

## Technical comments

1. There is some repetition of code, but the underlying architecture is like that, and I didn't have time for any serious redesign.
2. The commented out lines I left in the code are on purpose, they offer some extra debugging possibilities which people can use when troubleshooting.

## Types of changes


### Final checklist
